agent/host.go

```diff
@@ -52,6 +54,59 @@ type CPUInfo struct {
 	Microcode  string `json:"Microcode,omitempty"`
 }
 
+// Extract class and ID from the children node
+func extractClassAndID(children map[string]interface{}) (string, string) {
+	var foundID string
+	var foundClass string
+	for objectName, objectValue := range children {
+		switch objectValue.(type) {
+		case string:
+			if objectName == "class" {
+				foundClass = objectValue.(string)
+			}
+			if objectName == "id" {
+				foundID = objectValue.(string)
+			}
+			// Once we discover the class and ID, let's return them
+			if (len(foundID) > 0) && (len(foundClass) > 0) {
+				return foundClass, foundID
+			}
+		}
+	}
+	return "", ""
+}
+
+// Parse the LSHW data structure to rename children nodes
+func parseLshw(items map[string]interface{}) {
+	// For each object of the LSHW structure
+	for objectName, objectValue := range items {
+		switch objectValue.(type) {
+		case []interface{}:
+			// Only consider object that have children
+			if objectName != "children" {
+				return
+			}
+			// When we have children
+			for _, child := range objectValue.([]interface{}) {
+				// For each child
+				class, ID := extractClassAndID(child.(map[string]interface{}))
+				// Extract from the child the class and ID to rename it
+				if len(class) > 0 {
+					if items[class] == nil {
+						items[class] = []map[string]interface{}{}
+					}
+					// Let's create a new child with the name of the class and add an entry with the ID and point it to the actual child
+					items[class] = append(items[class].([]map[string]interface{}), map[string]interface{}{ID: child})
+					// Delete the actual entry so we renamed it
+					delete(items, objectName)
```

Comment:
Maybe id and class could be removed too from the final structure, it seems to be useless after transformation ?

Reply:
I was wondering the same. I though that keeping the ID/CLASS as members of a child might help some to get all needed info at the same "level" of the tree. Make sense ?

Reply:
I would prefer to remove them as it will reduce the amount of data in the graph/data store and the size of the messages. If needed we could simply introduce a custom marshal function to add them on fly or even re-introduce them ?